src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -39,6 +39,11 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
   }
 
   private def createVertex(workspace: Vertex, workspaceNamespace: String, workspaceName: String, entity: Entity, txn: RawlsTransaction): Vertex = txn withGraph { db =>
+    // check for illegal dot characters
+    validateUserDefinedString(entity.entityType)
+    validateUserDefinedString(entity.name)
+    entity.attributes.keys.foreach(validateUserDefinedString)
+
```

Comment:
The code block is here because we create entities in two places - ```save``` and ```cloneTheseEntities```. You could argue that if the entities already exist in another workspace, they've already been validated, but that seems like a silly reason to not do it.

We validate ```entityType``` and ```name``` because the story says to. It seems reasonable, and it might save us sometime in the future. It limits users by a slight amount, but they're already forbidden from using dots in attribute keys, so we might as well discourage them from using dots at all.